### ISSUE TITLE
Support for Vivaldi 

### ISSUE THREAD

**Issue by MrLordy** [xBrowserSync/App#6]
Support for the vivaldi browser would be really nice.  and a great addition :)

**Comment by gabrielcossette** [comment]
+1

**Comment by LeBaux** [comment]
Currently, there is no nice way to sync bookmarks from vivaldi to pretty much any mobile browser. Might be a good addition.

**Comment by nero120** [comment]
@LeBaux I agree. Last I checked the Vivaldi team had not defined the API for their bookmarking system, and there's still no documentation for it. For simpler web extensions it's probably ok just to try it and see, but xBrowserSync is pretty complex and makes heavy use of various web extension APIs, so the Vivaldi team really need to get that documentation out before I can add support for their browser.

**Comment by LeBaux** [comment]
Thanks for clarification. Bookmark syncing always baffled my mind and it is recurring issue across the browsers and time. Yet you can export/import simple HTML with ease (and it contains all the data 99% user needs). I understand that this thing is more complex than that and without proper documentation you can't do much. I searched the forums and it seems like the issue of bookmarks (not only API) is unresolved for more than 2 years. Vivaldi is currently focused heavily on building their own sync, but this still doesn't solve sync to mobile and I doubt Vivaldi mobile will be made in foreseeable future. So xBrowserSync might fill this hole nicely when their API comes (if ever, but I suspect it will come with their native sync).

**Comment by nero120** [comment]
Yeah it's really annoying that new browsers aren't fully supporting the web extension API, which means many extensions won't work with their browsers - let's face it, most devs aren't going to build specific versions of their extensions for fringe browsers!

Mozilla has been working for the past few years to [fully support](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Browser_support_for_JavaScript_APIs) the web extensions API, Brave has [poor support](https://github.com/brave/browser-laptop/wiki/Extensions-API-support), and as mentioned before Vivaldi don't even have a page indicating compatibility (even though the browser is a fork of Chromium!).

My goal in the short term is to support the major desktop browsers that fully implement the web extensions API (Firefox next, perhaps Edge) so that xBrowserSync users can access their bookmarks from any of these browsers, and on their mobiles using the xBrowserSync mobile app.

**Comment by nergdron** [comment]
I didn't know this bug existed, and setup the existing chrome extension on vivaldi today. it seems to work fine, including bookmarks bar sync. so you may be able to just close this. ;)

**Comment by nero120** [comment]
@nergdron whilst xBrowserSync may appear to work for the most part on any Chrome/Firefox based browsers they're not officially supported until they are. Vivaldi support will come with v1.6.0 so until then I won't be able to resolve any issues raised, but hopefully you won't encounter any! 😄

**Comment by clemthi** [comment]
Just tested on version 3.0 : it's somehow working.
Apart from the bug raised in #159 , the main issue is that only the toolbar bookmarks are included within Vivaldi.

**Comment by nero120** [comment]
Interesting, thanks for the info @clemthi.

**Comment by mohsenkhanpour** [comment]
@nero120 Thanks for the extension.
After testing this extension on Vivaldi, I found out that when syncing xBrowserSync deletes every bookmark, and Vivaldi moves it to Trash. If you delete the Trash it triggers a sync and after sync there are still items in the Trash.
I think if Trash and Speed Dial are not synced on Vivaldi, it would make compatibility easier, as everything else works as expected.

**Comment by sunflower-seed** [comment]
Maybe it helps...:

Vivaldi  seems to store its bookmarks as json in the local user folder.
Eg. under linux
`~/.config/vivaldi/Default/Bookmarks`

where 'Default' seems to be the user profile.

**Comment by nero120** [comment]
@sunflower-seed xBrowserSync uses the [web extension APIs](https://developer.chrome.com/docs/extensions/reference/) to access browser data, so how the browser chooses to store data at a lower level makes no difference to xBrowserSync.

**Comment by i2** [comment]
@nero120 Has there been any changes regarding adding support for Vivaldi? Thank you!

**Comment by nohart** [comment]
> @nero120 Thanks for the extension. After testing this extension on Vivaldi, I found out that when syncing xBrowserSync deletes every bookmark, and Vivaldi moves it to Trash. If you delete the Trash it triggers a sync and after sync there are still items in the Trash. I think if Trash and Speed Dial are not synced on Vivaldi, it would make compatibility easier, as everything else works as expected.

I have the exact same issue. I just installed the plugin and synced between Chrome and Vivaldi and if I empty the trash it just loads back up with all the bookmarks.

**Comment by git-rz** [comment]
After a recent update of Vivaldi (6.1.3035.75 (Stable channel) (x86_64) ), this extension now causes Vivaldi to crash upon startup.

I had to uninstall it by first finding it in the Extensions folder, and then remove it:
![image](https://github.com/xbrowsersync/app/assets/53906153/64685255-e2e7-444e-a0bb-d3e38738f50d)

Vivaldi would then start properly. I was able to finish the removal by going to vivaldi:extensions to remove the left-over config.

This is a sad day. The limited support is gone and I have to find an alternative. Floccus looks promising.

**Comment by abudden** [comment]
> After a recent update of Vivaldi (6.1.3035.75 (Stable channel) (x86_64) ), this extension now causes Vivaldi to crash upon startup.
> 
> I had to uninstall it by first finding it in the Extensions folder, and then remove it: ![image](https://user-images.githubusercontent.com/53906153/245212140-64685255-e2e7-444e-a0bb-d3e38738f50d.png)
> 
> Vivaldi would then start properly. I was able to finish the removal by going to vivaldi:extensions to remove the left-over config.
> 
> This is a sad day. The limited support is gone and I have to find an alternative. Floccus looks promising.

I'd mostly given up on xbrowsersync ever working again, but thank you for telling me about floccus.  I just installed it and it was very easy to set up and (so far) seems to work very well.
